### ISSUE TITLE
ULTRA CI Tests Failing

### ISSUE THREAD

**Issue by christianjans** [huawei-noah/SMARTS#708]
ULTRA CI tests, specifically the ones that use multi-agent functionality in the EvaluateTest seem to be failing.

**Comment by christianjans** [comment]
Unable to SSH into my machine... A solution is ready but not public. If needed, simply just comment out the failing tests and we will deal with implementing the solution later.

**Comment by JenishPatel99** [comment]
@christianjans It seems that the fix in PR #709 is working. However, a few commits later the evaluate test (CI) failed due. The error log is shown below

```python
============================= test session starts ==============================
platform linux -- Python 3.7.9, pytest-6.2.2, py-1.10.0, pluggy-0.13.1 -- /__w/SMARTS/SMARTS/ultra/.venv/bin/python3.7
cachedir: .pytest_cache
benchmark: 3.2.3 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
rootdir: /__w/SMARTS/SMARTS/ultra
plugins: benchmark-3.2.3, cov-2.11.1, notebook-0.6.1, forked-1.3.0, xdist-2.2.1
collecting ... collected 39 items

tests/test_adapter.py::AdapterTest::test_observation_features PASSED     [  2%]
tests/test_adapter.py::AdapterTest::test_rewards_adapter PASSED          [  5%]
tests/test_adapter.py::AdapterTest::test_rewards_returns PASSED          [  7%]
tests/test_analysis.py::AnalysisTest::test_analyze_scenario PASSED       [ 10%]
tests/test_analysis.py::AnalysisTest::test_interface_analyze PASSED      [ 12%]
tests/test_analysis.py::AnalysisTest::test_save_histogram PASSED         [ 15%]
tests/test_env.py::EnvTest::test_headless PASSED                         [ 17%]
tests/test_env.py::EnvTest::test_scenario_task PASSED                    [ 20%]
tests/test_env.py::EnvTest::test_timestep_sec PASSED                     [ 23%]
tests/test_episode.py::EpisodeTest::test_episode_counter PASSED          [ 25%]
tests/test_episode.py::EpisodeTest::test_episode_record PASSED           [ 28%]
tests/test_evaluate.py::EvaluateTest::test_a_folders PASSED              [ 30%]
tests/test_evaluate.py::EvaluateTest::test_evaluate_agent PASSED         [ 33%]
tests/test_evaluate.py::EvaluateTest::test_evaluate_cli PASSED           [ 35%]
tests/test_evaluate.py::EvaluateTest::test_evaluate_cli_multiagent PASSED [ 38%]
tests/test_evaluate.py::EvaluateTest::test_evaluate_multiagent PASSED    [ 41%]
tests/test_evaluate.py::EvaluateTest::test_evaluation_check PASSED       [ 43%]
:display:x11display(fatal): X fatal error on display 0x555d9a3b7060
tests/test_evaluate.py::EvaluateTest::test_evaluation_check_multiagent argv[0]=
argv[0]=
*** Aborted at 1616627621 (unix time) try "date -d @1616627621" if you are using GNU date ***
PC: @                0x0 (unknown)
*** SIGABRT (@0x124) received by PID 292 (TID 0x7f5a1a815740) from PID 292; stack trace: ***
    @     0x7f5a1a3ed980 (unknown)
    @     0x7f5a19434fb7 gsignal
    @     0x7f5a19436921 abort
    @     0x7f5a04bf96a1 google::LogMessage::Flush()
    @     0x7f5a04bf9771 google::LogMessage::~LogMessage()
    @     0x7f5a04bacbe9 ray::RayLog::~RayLog()
    @     0x7f5a0486392d ray::CoreWorkerProcess::~CoreWorkerProcess()
    @     0x7f5a048639da std::unique_ptr<>::~unique_ptr()
    @     0x7f5a19439161 (unknown)
    @     0x7f5a1943925a exit
    @     0x7f5924275968 _XIOError
    @     0x7f5924273a41 _XReply
    @     0x7f592426f18d XSync
    @     0x7f592424fe1e XCloseDisplay
    @     0x7f59341e81f7 (unknown)
    @     0x7f593424d62e (unknown)
    @     0x7f5998d820d9 (unknown)
    @     0x555d92f8624f (unknown)
    @     0x555d92fa07eb (unknown)
    @     0x555d92fc03fe (unknown)
    @     0x555d92fa09d8 (unknown)
    @     0x555d92fc03fe (unknown)
    @     0x555d92fa0a35 (unknown)
    @     0x555d92fc1f52 (unknown)
    @     0x555d93026e32 (unknown)
    @     0x555d930bd2cd (unknown)
    @     0x555d93026785 _PyObject_GC_Malloc
    @     0x555d92fbeff5 PyType_GenericNew
    @     0x555d930749a8 (unknown)
    @     0x555d9307389e (unknown)
    @     0x555d930739a7 (unknown)
    @     0x555d930738d9 (unknown)
Aborted (core dumped)
```
I ran the evaluate test on the remote machine (not CI) and it does work. I am not really sure why this is happening, nevertheless I will try to push a dry commit to see if the error is produced again
```

**Comment by christianjans** [comment]
Addressed the above comment offline with @JenishPatel99, it is unrelated to the changes made in this PR.

**Comment by christianjans** [comment]
Closed with the merge of #709.
